### ISSUE TITLE
Compatibility with Godot 3.

### ISSUE THREAD

**Issue by Malkverbena** [TheHX/add_primitives#16]
Could you make this pluging compatible with Godot 3?

**Comment by xemjeff** [comment]
I am also trying to get this working in Godot 3.  Using Godot-3.0.2-stable, after I pulled this plugin from git and set up the sub-folder "res://addons/add_primitives"  I get these errors when switching the plugin to "Active" in Project Settings:

```
 res://addons/add_primitives/src/AddPrimitives.gd:170 - Parse Error: ':' expected at end of line.
 modules\gdscript\gdscript.cpp:583 - Method/Function Failed, returning: ERR_PARSE_ERROR
 res://addons/add_primitives/Plugin.gd:60 - Invalid call. Nonexistent function 'new' in base 'GDScript'.
```
It looks like the "new" is called in Plugin.gd (line 6) but AddPrimitives.gd is failing on the if statement below - but I can see that the line ends in a :  (colon). 

```
	for c in popup_menu.get_children():
		if c extends PopupMenu:      # <---- this line gives a parsing error
			c.free()
```
Any idea as to why we get a parse error here?

**Comment by kubecz3k** [comment]
`extends` keyword has changed its meaning as far as I know you can use it only in the first line of the script to set the inheritance hierarchy. In this case instead `is` keyword should be used.

**Comment by Malkverbena** [comment]
@xemjeff   Did you make any progress?

**Comment by xemjeff** [comment]
Yes, some. I found that the syntax and the way that menus work in 3.0 are different. So I wrote my own script to handle the menus and create the shapes, but I used the primitives already in the TheHX code.  My version does not have the editor panels to change the shapes, but it can create the basic shapes.

![shapes3d](https://user-images.githubusercontent.com/8524620/40591712-0a2af870-61e4-11e8-9612-cc6bdc8376c2.png)

**Comment by Malkverbena** [comment]
Nice. I was think in port the whole code to 3.0. But I haven't enough for that.
Another ideia is rewrite it in C++, as a module.

**Comment by MaGaRCaN** [comment]
Any updates? It's one of the things I miss more from Unity, and have this feature in the 3.1 version would be awesome.
